Adaptive localization fails if there is only one observation.

**Describe the bug**
Running ert with adaptive localization and one observation crashes the update, with:
```
File .../iterative_ensemble_smoother/experimental.py", line 276, in assimilate
    assert cov_YY.ndim == 2, "'cov_YY' must be a 2D array"
AssertionError: 'cov_YY' must be a 2D array
```

**To reproduce**
Steps to reproduce the behaviour:
1. Modify poly example to only have one observation
2. `ert gui poly.ert`
3. Run ES (or ES-MDA) with localization activated

**Expected behaviour**
Would expect it to pass the analysis step.

--- a/src/ert/analysis/_es_update.py
+++ b/src/ert/analysis/_es_update.py
@@ -472,7 +472,7 @@ def analysis_ES(
         )
 
         # Pre-calculate cov_YY
-        cov_YY = np.cov(S)
+        cov_YY = np.atleast_2d(np.cov(S))
 
         D = smoother_adaptive_es.perturb_observations(
             ensemble_size=ensemble_size, alpha=1.0

--- a/tests/integration_tests/analysis/test_adaptive_localization.py
+++ b/tests/integration_tests/analysis/test_adaptive_localization.py
@@ -58,6 +58,42 @@ def test_that_adaptive_localization_with_cutoff_1_equals_ensemble_prior():
     assert np.allclose(posterior_sample, prior_sample)
 
 
+@pytest.mark.usefixtures("copy_poly_case", "using_scheduler")
+def test_that_adaptive_localization_works_with_a_single_observation():
+    """This is a regression test as ert would crash if adaptive localization
+    was run with a single observation.
+    """
+    set_adaptive_localization_0 = dedent(
+        """
+        ANALYSIS_SET_VAR STD_ENKF LOCALIZATION True
+        ANALYSIS_SET_VAR STD_ENKF LOCALIZATION_CORRELATION_THRESHOLD 0.0
+        """
+    )
+
+    with open("poly.ert", "r+", encoding="utf-8") as f:
+        lines = f.readlines()
+        lines.insert(9, set_adaptive_localization_0)
+
+    content = """GENERAL_OBSERVATION POLY_OBS {
+        DATA       = POLY_RES;
+        INDEX_LIST = 0;
+        OBS_FILE   = poly_obs_data.txt;
+    };"""
+
+    with open("observations", "w") as file:
+        file.write(content)
+
+    content = "2.1457049781272213 0.6"
+
+    with open("poly_obs_data.txt", "w") as file:
+        file.write(content)
+
+    with open("poly_localization_0.ert", "w", encoding="utf-8") as f:
+        f.writelines(lines)
+
+    _, _ = run_cli_ES_with_case("poly_localization_0.ert")
+
+
 @pytest.mark.integration_test
 @pytest.mark.usefixtures("copy_poly_case", "using_scheduler")
 def test_that_adaptive_localization_with_cutoff_0_equals_ESupdate():

